queue/QueueConsumer.js

```diff
@@ -43,12 +115,30 @@ module.exports = class QueueConsumer extends BaseQueue {
     // messages that are being handled at once)
     await channel.prefetch(this._prefetchCount)
 
-    // TODO: other assertions may need ot be done on the queue
-    await channel.assertQueue(queueName, {
+    // create exchange for dead letter exchange,
+    // we specify the "fanout" so that messages published to
+    // this exchange sent to all queues that are bound
+    // to the exchange.
+    //
+    // In this case, we only have a single queue to listening
+    // for this exchange
+    await channel.assertExchange(deadLetterExchangeName, AMQ_FANOUT)
+
+    // assert the main queue to consume from
+    await assertQueue(channel, queueName)
```

Comment:
I force the `QueueProducer` and `QueueConsumer` to use the same `assertQueue` function so that we can keep errors them pointed at the same queues. Otherwise any differences from the assertions could cause either one to fail to start.